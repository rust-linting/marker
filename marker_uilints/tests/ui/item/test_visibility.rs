fn test_vis_private() {}

pub fn test_vis_public() {}

mod module {
    pub(crate) fn test_vis_pub_crate() {}

    pub(super) fn test_vis_pub_super_crate_root() {}

    mod nested {
        pub(super) fn test_vis_pub_super() {}

        pub(in crate::module) fn test_vis_pub_in_path() {}
    }
}

fn main() {}
