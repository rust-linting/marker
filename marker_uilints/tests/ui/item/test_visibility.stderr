warning: can you see this item?
 --> $DIR/test_visibility.rs:1:4
  |
1 | fn test_vis_private() {}
  |    ^^^^^^^^^^^^^^^^
  |
  = note: vis.is_pub()                  -> false
  = note: vis.is_pub_with_path()        -> false
  = note: vis.is_pub_crate()            -> false
  = note: vis.is_explicit()             -> false
  = note: vis.pub_with_path_module_id() -> None
  = note: vis.module_id()               -> Some(ItemId(..))
  = note: vis.span(): `None`
  = note: `#[warn(marker::marker_uilints::test_item_visibility)]` on by default

warning: can you see this item?
 --> $DIR/test_visibility.rs:3:8
  |
3 | pub fn test_vis_public() {}
  |        ^^^^^^^^^^^^^^^
  |
  = note: vis.is_pub()                  -> true
  = note: vis.is_pub_with_path()        -> false
  = note: vis.is_pub_crate()            -> false
  = note: vis.is_explicit()             -> true
  = note: vis.pub_with_path_module_id() -> None
  = note: vis.module_id()               -> None
  = note: vis.span(): `Some("pub")`

warning: can you see this item?
 --> $DIR/test_visibility.rs:6:19
  |
6 |     pub(crate) fn test_vis_pub_crate() {}
  |                   ^^^^^^^^^^^^^^^^^^
  |
  = note: vis.is_pub()                  -> false
  = note: vis.is_pub_with_path()        -> true
  = note: vis.is_pub_crate()            -> true
  = note: vis.is_explicit()             -> true
  = note: vis.pub_with_path_module_id() -> Some(ItemId(..))
  = note: vis.module_id()               -> Some(ItemId(..))
  = note: vis.span(): `Some("pub(crate)")`

warning: can you see this item?
 --> $DIR/test_visibility.rs:8:19
  |
8 |     pub(super) fn test_vis_pub_super_crate_root() {}
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: vis.is_pub()                  -> false
  = note: vis.is_pub_with_path()        -> true
  = note: vis.is_pub_crate()            -> true
  = note: vis.is_explicit()             -> true
  = note: vis.pub_with_path_module_id() -> Some(ItemId(..))
  = note: vis.module_id()               -> Some(ItemId(..))
  = note: vis.span(): `Some("pub(super)")`

warning: can you see this item?
  --> $DIR/test_visibility.rs:11:23
   |
11 |         pub(super) fn test_vis_pub_super() {}
   |                       ^^^^^^^^^^^^^^^^^^
   |
   = note: vis.is_pub()                  -> false
   = note: vis.is_pub_with_path()        -> true
   = note: vis.is_pub_crate()            -> false
   = note: vis.is_explicit()             -> true
   = note: vis.pub_with_path_module_id() -> Some(ItemId(..))
   = note: vis.module_id()               -> Some(ItemId(..))
   = note: vis.span(): `Some("pub(super)")`

warning: can you see this item?
  --> $DIR/test_visibility.rs:13:34
   |
13 |         pub(in crate::module) fn test_vis_pub_in_path() {}
   |                                  ^^^^^^^^^^^^^^^^^^^^
   |
   = note: vis.is_pub()                  -> false
   = note: vis.is_pub_with_path()        -> true
   = note: vis.is_pub_crate()            -> false
   = note: vis.is_explicit()             -> true
   = note: vis.pub_with_path_module_id() -> Some(ItemId(..))
   = note: vis.module_id()               -> Some(ItemId(..))
   = note: vis.span(): `Some("pub(in crate::module)")`

warning: 6 warnings emitted

